File: Python/ExpenseManager.py
class ExpenseManager:
    
    def __init__(self):
        self.income = 0
        self.withdraw = 0
        self.balance = 0

    def income_update(self, income):
        self.income = income
        self.update_balance()
    
    def withdraw_update(self, amount):
        if amount <= self.balance:
            self.withdraw += amount
            self.update_balance()

        else:
            print("Insufficient fund!")

    def update_balance(self):
        self.balance = self.income - self.withdraw

    def get_balance(self):
        return self.balance
    

em = ExpenseManager()

income = int(input("enter the income: "))
em.income_update(income)

withdraw = int(input("enter the withdraw amount: "))
em.withdraw_update(withdraw)

print("balance after withdraw " + str(em.get_balance()))

File: Java/ExpenseManager.java
package Java;
import java.util.Scanner;

class ExpenseManager{

    double income, withdraw, balance;

    public void incomeUpdate(double income){
        this.income = income;
        balanceUpdate();
    }

    public void withdrawUpdate(double amount){
        if(balance >= amount){
            this.withdraw += amount;
            balanceUpdate();
        }

        else{
            System.out.println("Insufficient fund!");
        }
    }

    public void balanceUpdate(){
        this.balance = income - withdraw;
    }

    public double getBalance(){
        return balance;
    }

    public static void main(String[] args) {

        Scanner scan = new Scanner(System.in);

        ExpenseManager em = new ExpenseManager();

        System.out.print("Enter the income amount: ");
        double income = scan.nextDouble();
        em.incomeUpdate(income);

        System.out.print("Enter the withdraw amount: ");
        double withdraw = scan.nextDouble();
        em.withdrawUpdate(withdraw);

        System.out.print("Amount after withdraw = " + em.balance);

        scan.close();
    }
}